### PR TITLE
Rebase Automation Script

### PR DESCRIPTION
**Pull request by tjungblu**
@josefkarasek and I wanted to share the rebase script we've recently used for the 4.6 rebase in #1035 

We're going to work on automating this a bit more, especially around the conflict resolution and a bot after the feature freeze. We'll share a design doc with you soon (cc @sherine-k).

/kind feature
/label tide/merge-method-squash

### CODE REVIEW

File: openshift-hack/rebase.sh [hunk 1]
```diff
@@ -0,0 +1,171 @@
...
+  # TODO(tjungblu): verify that the conflicts have been resolved
+  git commit -am "UPSTREAM: <drop>: manually resolve conflicts"
+fi
+
+# update dependencies
+sed -i -E "/=>/! s/(\tgithub.com\/openshift\/[a-z|-]+) (.*)$/\1 $openshift_release/" go.mod
+go mod tidy
+
+hack/update-vendor.sh
+# shellcheck disable=SC2181
```
Comment (tjungblu) [inline openshift-hack/rebase.sh:131]: go mod and the vendor update can potentially take a higher go version locally, we need to use podman for the invocation too.

Comment (soltysh) [reply openshift-hack/rebase.sh:131]: Yeah, `podman run -it --rm -v $( pwd ):/go/k8s.io/kubernetes:Z --workdir=/go/k8s.io/kubernetes registry.ci.openshift.org/openshift/release:rhel-8-release-golang-1.16-openshift-4.9 make update OS_RUN_WITHOUT_DOCKER=yes` with appropriate golang and openshift release is the best option you have as mentioned elsewhere in this doc.

Comment (soltysh) [reply openshift-hack/rebase.sh:131]: Oh, I misread, yeah, for vendor update maybe it's also worth using the same podman invocation as for `make update`.

Comment (tjungblu) [reply openshift-hack/rebase.sh:131]: yep, exactly. This is being done below a couple of lines, but that's how the build tags remained ultimately.

**Review by soltysh — APPROVED** [review]
/lgtm
/approve
/label backports/validated-commits
/remove-label backports/unvalidated-commits

**Pull request merged**